### PR TITLE
[RFC] Add `self.lr_schedulers()` to LightningModule for manual optimization

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Part of #6379.

Before disabling `lr_scheduler.step()` in manual optimization in #6379, this PR adds `self.lr_schedulers()` so that users can `lr_scheduler.step()` in LightningModule at arbitrary intervals in manual optimization.

Example:
```python
class Model(LightningModule):
    def __init__(self):
        self.automatic_optimization = False

    def training_step(self, batch, batch_idx):
        # single scheduler
        scheduler = self.lr_schedulers()

        # multiple schedulers
        scheduler1, scheduler2 = self.lr_schedulers()
```

### TODO
- [n/a] ~Update the docs~ I'll update the docs in the following PR which disables lr_scheduler.step() in manual optimization because this PR itself doesn't enable users to call `step()` in manual optimization.
- [x] Add a test

## Before submitting
- [RFC] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

Related to #6825.

### CODE REVIEW

File: pytorch_lightning/core/lightning.py [hunk 1]
```diff
@@ -119,6 +119,20 @@ def optimizers(self, use_pl_optimizer: bool = True) -> Union[Optimizer, List[Opt
         # multiple opts
         return opts
 
+    def lr_schedulers(self) -> Optional[Union[Any, List[Any]]]:
+        if not self.trainer.lr_schedulers:
+            return None
+
+        # ignore other keys "interval", "frequency", etc.
+        lr_schedulers = [s["scheduler"] for s in self.trainer.lr_schedulers]
```
Comment (akihironitta) [inline pytorch_lightning/core/lightning.py:127]: Even when dict keys like `"interval"`, `"monitor"`, are defined in `configure_optimizers()`, this line ignores all of the keys except `"scheduler"`. [[lr scheduler docs](https://pytorch-lightning.readthedocs.io/en/latest/common/optimizers.html#learning-rate-scheduling)]

**Review by carmocca — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
Looks neat !

**Review by awaelchli — APPROVED** [review]

**Comment by maxoppelt** [comment]
I think this update introduced a new bug:

When automatic_optimization is disabled and you are using schedulers that require a metric, e.g. th.optim.lr_scheduler.ReduceLROnPlateau, an error is raised:

`pytorch_lightning.utilities.exceptions.MisconfigurationException: The lr scheduler dict must include a monitor when a "ReduceLROnPlateau" scheduler is used. For example: {"optimizer": optimizer, "lr_scheduler": {"scheduler": scheduler, "monitor": "your_loss"}}
`

As soon as you define a monitor you get a warning:

`/pytorch_lightning/utilities/distributed.py:69: RuntimeWarning: The lr scheduler dict contains the key(s) ['monitor'], but the keys will be ignored. You need to call "lr_scheduler.step()" manually in manual optimization.
`

Pytorch Lightning Version 1.3.1

**Comment by maxoppelt** [comment]
Yes, that is a possible solution. Disable the raise of MisconfigurationException when using automatic_optimization False.

Another design choice could be: Disable the warning and provide access to the monitor key in training_epoch_end/validation_epoch_end.

Minor remark on the documentation: https://pytorch-lightning.readthedocs.io/en/latest/common/optimizers.html#learning-rate-scheduling-manual is misleading: Most schedulers have an epoch argument in the step method. Therefore one should not call scheduler.step() in training_step(). Especially when adding epoch as argument to your scheduler step: You get an EPOCH_DEPRECATION_WARNING. 

This could lead to misunderstandings, when reading the doc. However calling the scheduler in training_epoch_end() might be problematic when using multi dataloaders or ddp training?

**Comment by awaelchli** [comment]
> Disable the raise of MisconfigurationException when using automatic_optimization False.

My preference.

> Another design choice could be: Disable the warning and provide access to the monitor key in training_epoch_end/validation_epoch_end.

There is already a pattern for this, by returning the value in the step method or by using torchmetrics.
So I think we don't need another way. Or what would be concretely your suggestion?

> However calling the scheduler in training_epoch_end() might be problematic when using multi dataloaders or ddp training?

For multi dataloaders, training_epoch_end() will receive outputs for all. So I see no big problem here.
In DDP training we update the scheduler in each process but when a metric is required, we probably want to update with the same metric value in all process. We need to think of something here with minimal code changes required for the user.

Are you interested sending a PR, for the error message handling / doc improvements?
